src/data_hub/plotting.py

```diff
@@ -64,65 +101,212 @@ def scatter(
     )
     p.toolbar.logo = None
 
-    p.scatter(
+    # make scatter plot of data
+    scatter_glyphs = p.scatter(
         x="x",
         y="y",
-        source=source,
+        source=scatter_source,
         size="size",
         color="blue",
         alpha=0.6,
     )
 
+    # draw the marker
+    p.scatter(
+        x="x",
+        y="y",
+        size=10,
+        fill_color=None,
+        color="red",
+        source=marker_source,
+        line_width=3,
+    )
+
     # Set initial axis labels
-    p.xaxis.axis_label = initial_x
-    p.yaxis.axis_label = initial_y
+    p.xaxis.axis_label = x
+    p.yaxis.axis_label = y
 
     # Create dropdown selectors for X and Y axes
     x_select = Select(
         title="X-Axis",
-        value=initial_x,
-        options=cols,
+        value=x,
+        options=numeric_cols,
         width=select_width,
     )
     y_select = Select(
         title="Y-Axis",
-        value=initial_y,
-        options=cols,
+        value=y,
+        options=numeric_cols,
         width=select_width,
     )
     size_select = Select(
         title="Size",
-        value=initial_size,
-        options=cols,
+        value=size,
+        options=numeric_cols,
         width=select_width,
     )
 
-    # JavaScript callback to update data, axis labels, and point sizes on select change
+    # create dropdown selectors for label column
+    if label_data:
+        first_col = list(label_data.keys())[0]
+        label_column_select = Select(
+            title="Label Column",
+            value=first_col,
+            options=list(label_data.keys()),
+            width=select_width,
+        )
+
+        label_select = Select(
+            title="Label",
+            value=label_data[first_col][0],
+            options=list(set(label_data[first_col])),
+            width=select_width,
+        )
```

Comment:
interactive clustering only supported if there's a select column in the database